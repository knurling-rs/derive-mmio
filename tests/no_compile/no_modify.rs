#[derive(derive_mmio::Mmio)]
#[repr(C)]
struct Uart {
    // this is a FIFO register, so modifying it does not really make sense.
    #[mmio(Read, Write)]
    fifo: u32,
}

fn main() {
    let mut uart = Uart { fifo: 0x0 };

    // Safety: We're pointing at a real object
    let mut mmio_uart = unsafe { Uart::new_mmio(core::ptr::addr_of_mut!(uart)) };
    mmio_uart.modify_uart();
}
