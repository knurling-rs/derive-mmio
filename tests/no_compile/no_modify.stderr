error[E0599]: no method named `modify_uart` found for struct `MmioUart` in the current scope
  --> tests/no_compile/no_modify.rs:14:15
   |
1  | #[derive(derive_mmio::Mmio)]
   |          ----------------- method `modify_uart` not found for this struct
...
14 |     mmio_uart.modify_uart();
   |               ^^^^^^^^^^^ method not found in `MmioUart<'_>`
